mantle/kola/tests/misc/network.go

```diff
@@ -372,6 +382,52 @@ var (
 `
 )
 
+func NetworkBondWithRestart(c cluster.TestCluster) {
+	primaryMac := "52:55:00:d1:56:00"
+	secondaryMac := "52:55:00:d1:56:01"
+	primaryIp := "192.168.0.55"
+	secondaryIp := "192.168.0.56"
+
+	setupBondWithDhcpTest(c, primaryMac, secondaryMac, primaryIp, secondaryIp)
+
+	m := c.Machines()[0]
+	expectedUpConnections := getOvsRelatedConnections(c, m)
+	err := checkConnectionsUp(c, m, expectedUpConnections)
+	if err != nil {
+		c.Fatalf("connections check failed before reboot: %v", err)
+	}
+
+	err = m.Reboot()
+	if err != nil {
+		c.Fatalf("failed to reboot the machine: %v", err)
+	}
+
+	err = checkConnectionsUp(c, m, expectedUpConnections)
+	if err != nil {
+		c.Fatalf("connections check failed post reboot: %v", err)
+	}
+}
+
+func getOvsRelatedConnections(c cluster.TestCluster, m platform.Machine) []string {
+	return grepConnections(c, m, "ovs")
+}
+
+func grepConnections(c cluster.TestCluster, m platform.Machine, subString string) []string {
+	output := string(c.MustSSH(m, fmt.Sprintf("nmcli con show | awk '{print $1}' | grep %s", subString)))
```

Comment:
I don't think this handles spaces correctly. Looks like `nmcli` supports querying fields, so this should work:

```
nmcli -g NAME con show | grep %s
```

Reply:
yes totally, thanks!